pkg/kubelet/util/pluginwatcher/plugin_watcher_test.go

```diff
@@ -238,11 +238,22 @@ func TestPlugiRegistrationFailureWithUnsupportedVersionAtKubeletStart(t *testing
 	hdlr := NewExampleHandler(supportedVersions, false /* permitDeprecatedDir */)
 	hdlr.AddPluginName(pluginName)
 
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		require.True(t, waitForEvent(t, exampleEventValidate, hdlr.EventChan(p.pluginName)))
+		require.False(t, waitForPluginRegistrationStatus(t, p.registrationStatus))
+	}()
+
 	w := newWatcherWithHandler(t, hdlr, false /* testDeprecatedDir */)
 	defer func() { require.NoError(t, w.Stop()) }()
 
-	require.True(t, waitForEvent(t, exampleEventValidate, hdlr.EventChan(p.pluginName)))
-	require.False(t, waitForPluginRegistrationStatus(t, p.registrationStatus))
+	select {
+	case <-c:
+		return
+	case <-time.After(2 * time.Second):
```

Comment:
```suggestion
	case <-time.After(wait.ForeverTestTimeout):
```

You can't wait timeout before that in a unit test, as it will flake in the CI environment.

Reply:
Fixed. I took the chance and fixed the other `time.After` calls from this file.